src/EditorFeatures/Core/Implementation/SearchNuGet/AbstractOfferNuGetSearchCodeFixProvider.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editor.Implementation.CodeActions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Options.Providers;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text.Editor;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.SearchNuGet
+{
+    internal abstract class AbstractOfferNuGetSearchCodeFixProvider : CodeFixProvider
+    {
+        private readonly ILightBulbBroker _lightBulbBroker;
+
+        protected AbstractOfferNuGetSearchCodeFixProvider(ILightBulbBroker lightBulbBroker)
+        {
+            _lightBulbBroker = lightBulbBroker;
+        }
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var workspace = context.Document.Project.Solution.Workspace;
+            var options = workspace.Options;
+            var searchNuGet = options.GetOption(NuGetSearchOptions.SearchNuGet);
+
+            // Check if the user has enabled or disabled this feature.  If so, do nothing.
+            if (!searchNuGet.HasValue)
+            {
+                context.RegisterCodeFix(new MyCodeAction(workspace, _lightBulbBroker), context.Diagnostics);
+            }
+
+            return SpecializedTasks.EmptyTask;
+        }
+
+        private class MyCodeAction : CodeAction
+        {
+            private readonly Workspace _workspace;
+            private readonly ILightBulbBroker _lightBulbBroker;
+
+            public override string Title => "Search NuGet.org for types";
+
+            public MyCodeAction(Workspace workspace, ILightBulbBroker lightBulbBroker)
+            {
+                _workspace = workspace;
+                _lightBulbBroker = lightBulbBroker;
+            }
+
+            protected override Task<IEnumerable<CodeActionOperation>> ComputeOperationsAsync(CancellationToken cancellationToken)
+            {
+                return Task.FromResult(SpecializedCollections.SingletonEnumerable<CodeActionOperation>(
```

Comment:
I think it's much clearer personally.  "Wordy" isn't a negative in my assessment of code.